### PR TITLE
[reorg] new Recipes section; Basics streamlined

### PR DESCRIPTION
**Pull request by sarah11918**
Initial reorganization of the sidebar! :partying_face: 

tl/dr: 
- introduces a new Recipes section in sidebar for "guide landing pages"
- new "All Recipes" page as a landing page for some existing examples/walkthroughs
- shuffles pages in sidebar: fewer pages in basic (moved to guides); fewer pages in guides (moved to recipes)

This PR attempts to make the content of the Learn sidebar sections "a good length" so as not to be overwhelming.

___Getting Started:___
~~The CLI reference page has been renamed and moved here. Also, the introduction has been re-written for a "getting started" audience to introduce the `astro` commands they will use right from the beginning, and explains the `package.json` scripts that are included after running`create-astro` and how (and why) to use them.~~
- **OUT OF SCOPE**: CLI Reference page will remain where it is, but we know we want to have "friendlier CLI content earlier, in Learn somewhere, while still being able to leave the techical descriptions of each command for Reference.

___Basics:___
- Only a few pages kept here as truly "basics" you "need to know"
- Part of the idea is to make this section look "easy to manage/get through"
- **Big Change**: Astro components page - about half of the content has been removed:
   - New page "Astro Syntax" for the entire JSX-like expressions material
   - Script and style sections removed entirely as these are covered in their own pages
- **OUT OF SCOPE** : rewriting Markdown/MDX to remove MDX content and KISS; re-editing Routing page to incorporate more of SSR basics/examples

___Guides:___
- Some entire pages removed from sidebar (file not changed) and now linked to from "All Recipes"
- Some pages had their long, specific examples removed and into new pages in `/recipes/` for better discoverability

___Recipes:___
- New section header added (also cc'd i18n-gang to ask them to contribute translations for "recipes" and "all recipes" in their languages and to PR *this branch*. To date, two have done so.
- Moved all fancy/magic/landing pages HERE: CMS guides, Deploy Guides, Migrate to Astro guides, Integrations landing page
- Created an "All Recipes" page (is it really all, if those other pages aren't here? :sweat_smile: ) which will be a landing page for longer, walk-through, guided examples so they don't have to overwhelm the other content pages. In some cases, entire pages that seem to be "how to do something" (e.g. fonts) have been moved here.
- Edited our sidebar logic so that "All Recipes" in the sidebar is highlighted when viewing a recipe *that is located in `/recipes/`*.
- Created a new `<RecipeBox />` card component (based on the `<Box/>` we use in the tutorial) with its own svg icon to display the recipes. @delucis , go wild! I at least gave you a standalone component to work from.
- Created a new `<BackToRecipes />` component based on our current page navigation cards and tested it by throwing it on the "All Recipes" page (which does NOT need this component!)
-  (NB: made a new `/recipes/` folder in both `public/` and `/components/` to match the file structure of what goes on in the respective `/tutorial/` folders in each place)
- ~~**TO DO**~~ **DONE**: @delucis : 
   - I assume some kind of layout/nav specific to recipes so that they include this "Back to all recipes" component (or however you choose to do this) makes sense? I have not yet attempted any kind of layout for the individual recipes (that do not show up in the sidebar) and will let you continue that! 
   - Also, need to make sure existing guide pages (e.g. Fonts) show "All recipes" as highlighted when viewed. This can be done either by changing our existing logic, or moving the file (with appropriate redirect) into `/recipes/` because THAT logic is working fine.
   - Obviously, edit/replace my components as necessary for the cards! Make it look nice!
   - Anything else you encounter from a UI perspective when "using" the Recipes section!
 
___Reference Tab___:
- **ENTIRELY OUT OF SCOPE**

### CODE REVIEW

**Comment by Jutanium** [comment]
[Use UI Frameworks](https://deploy-preview-2550--astro-docs-2.netlify.app/en/core-concepts/framework-components/) stands out in the recipes section, as it's the only one that is formatted like a comprehensive guide/explainer and not a step-by-step how-to.

I think using UI frameworks is integral enough to Astro that it could go in the Basics section and isn't something we'd want to push down far in the nav.

**Comment by sarah11918** [comment]
Will think about this! 

We'll have to have more discussions about whether UI Frameworks are in fact "basics" because they are something that has to be installed. And, you don't *need* to know/use UI Frameworks, they *are* an add-on. In first discussions with Fred they were still in guides, but I wondered about moving them into recipes for discoverability. But you're right about them feeling maybe out of place in recipes?

(In fact, the Markdown AND MDX page is going to become just markdown, for a similar reason. But, since all I had done is edit the sidebar and not content in my first iteration, it still is the Markdown and MDX page.)

File: src/content/docs/en/reference/cli-reference.mdx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
-title: CLI Reference
+title: CLI Commands
```
Comment (Jutanium) [inline src/content/docs/en/reference/cli-reference.mdx:2]: The back half of this doc does feel "referencey." The `astro dev` command was already introduced in `Installation`, so I don't see the need to move the CLI reference into Getting Started. I'd rather explain the small amount they need to know about the CLI in `Installation` - the content you added is great, but it makes more sense there.

That said, it's NWTWWHB!

File: src/i18n/en/nav.ts [hunk 2]
```diff
@@ -12,72 +12,83 @@ export default [
...
 	{ text: 'Core Concepts', header: true, type: 'learn', key: 'coreConcepts' },
 	{ text: 'Why Astro', slug: 'concepts/why-astro', key: 'concepts/why-astro' },
 	{ text: 'MPA vs. SPA', slug: 'concepts/mpa-vs-spa', key: 'concepts/mpa-vs-spa' },
 	{ text: 'Astro Islands', slug: 'concepts/islands', key: 'concepts/islands' },
 
+	{ text: 'Tutorials', header: true, type: 'learn', key: 'tutorials' },
+	{ text: 'Build a Blog', slug: 'tutorial/0-introduction', key: 'blog-tutorial' },
+	// { text: 'Thinking with Islands', slug: 'tutorial/0-introduction', key: 'island-tutorial' },
+
 	{ text: 'Basics', header: true, type: 'learn', key: 'basics' },
```
Comment (Jutanium) [inline src/i18n/en/nav.ts:27]: I definitely agree with what you removed. To evaluate what remains, we need to flesh out the vision for Basics. What determines what you "need to know"? You can build a basic site without understanding anything but file-based routing, and certainly without understanding Markdown to the extent it's taught in the Markdown guide. `Pages` already covers the "basics" of both routing and Markdown, so I'd move both `Routing` and `Markdown & MDX` to `Guides`.

File: src/content/docs/en/core-concepts/astro-components.mdx [hunk 3]
```diff
@@ -4,11 +4,7 @@ description: An intro to the .astro component syntax.
 i18nReady: true
 ---
 
-**Astro components** are the basic building blocks of any Astro project. They are HTML-only templating components with no client-side runtime.
-
-**If you know HTML, you already know enough to write your first Astro component.**
-
-Astro component syntax is a superset of HTML. The syntax was [designed to feel familiar to anyone with experience writing HTML or JSX](#differences-between-astro-and-jsx), and adds support for including components and JavaScript expressions. You can spot an Astro component by its file extension: `.astro`.
+**Astro components** are the basic building blocks of any Astro project. They are HTML-only templating components with no client-side runtime. You can spot an Astro component by its file extension: `.astro`.
```
Comment (Jutanium) [inline src/content/docs/en/core-concepts/astro-components.mdx:7]: I know this wasn't introduced in this PR, but this sentence is misleading, as:
- "HTML-only" might sound like we're excluding CSS and JS, which can both be included in an Astro component
- You can include script tags for a client-side runtime

First thought:
```suggestion
**Astro components** are the basic building blocks of any Astro project. They are templating components that are built ahead of time, rather than on the client. You can spot an Astro component by its file extension: `.astro`.
```

File: src/content/docs/en/core-concepts/astro-components.mdx [hunk 4]
```diff
@@ -17,7 +13,7 @@ The most important thing to know about Astro components is that they **render to
 
 ## Component Structure
 
-An Astro component is made up of two main parts: the **Component Script** and the **Component Template**. Each part performs a different job, but together they aim to provide a framework that is both easy to use and expressive enough to handle whatever you might want to build.
+An Astro component is made up of two main parts: the **Component Script** and the **Component Template**. Each part performs a different job, but together they provide a framework that is both easy to use and expressive enough to handle whatever you might want to build.
```
Comment (Jutanium) [inline src/content/docs/en/core-concepts/astro-components.mdx:16]: Now is a good time to standardize to **frontmatter (script)**, which is used in the tutorial and more prevalent in our community.

File: src/content/docs/en/core-concepts/astro-components.mdx [hunk 5]
```diff
@@ -26,20 +22,6 @@ An Astro component is made up of two main parts: the **Component Script** and th
...
-  <Button title="Button 1" />
-  <Button title="Button 2" />
-  <Button title="Button 3" />
-</div>
-```
-
-
 ### The Component Script
 
 Astro uses a code fence (`---`) to identify the component script in your Astro component. If you've ever written Markdown before, you may already be familiar with a similar concept called *frontmatter.* Astro's idea of a component script was directly inspired by this concept.
```
Comment (Jutanium) [inline src/content/docs/en/core-concepts/astro-components.mdx:27]: Whether or not we change the name, I don't think the Markdown frontmatter comparison is helpful enough when they've heard of it to justify the cases when they haven't. They're more visually similar than conceptually similar.

File: src/content/docs/en/core-concepts/astro-components.mdx [hunk 6]
```diff
@@ -106,184 +88,21 @@ const myFavoritePokemon = [/* ... */];
 <p class:list={["add", "dynamic", {classNames: true}]} />
 ```
 
-## JSX-like Expressions
```
Comment (Jutanium) [inline src/content/docs/en/core-concepts/astro-components.mdx:109]: Totally agree with moving this into its own guide!

File: src/content/docs/en/core-concepts/astro-syntax.mdx [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+
+Astro component syntax is a superset of HTML. The syntax was [designed to feel familiar to anyone with experience writing HTML or JSX](#differences-between-astro-and-jsx), and adds support for including components and JavaScript expressions.
+
+
+## JSX-like Expressions
+
+You can define local JavaScript variables inside of the frontmatter component script between the two code fences (`---`) of an Astro component. You can then inject these variables into the component's HTML template using JSX-like expressions!
+
+:::note[dynamic vs reactive]
+Using this approach, you can include **dynamic** values that are calculated in the frontmatter.  But once included, these values are not **reactive** and will never change. Astro components are templates that only run once, at build time.
```
Comment (Jutanium) [inline src/content/docs/en/core-concepts/astro-syntax.mdx:17]: Whatever solution we come up with ("build or request time", "ahead of time", "build time or on the server") for the problem outlined in #2404 should be reflected here.
```suggestion
Using this approach, you can include **dynamic** values that are calculated in the frontmatter.  But once included, these values are not **reactive** and will never change. Astro components are templates that only run at build time or on the server.
```

File: src/content/docs/en/guides/data-fetching.mdx [hunk 8]
```diff
@@ -36,7 +36,7 @@ const randomUser = data.results[0];
 :::note
 Remember, all data in Astro components is fetched when a component is rendered.
 
-Your deployed Astro site will fetch data **once, at build time**. In dev, you will see data fetches on component refreshes. If you need to re-fetch data multiple times client-side, use a [framework component](/en/core-concepts/framework-components/) or a [client-side script](/en/core-concepts/astro-components/#client-side-scripts) in an Astro component.
+Your deployed Astro site will fetch data **once, at build time**. In dev, you will see data fetches on component refreshes. If you need to re-fetch data multiple times client-side, use a [framework component](/en/core-concepts/framework-components/) or a [client-side script](/en/guides/client-side-scripts/) in an Astro component.
```
Comment (Jutanium) [inline src/content/docs/en/guides/data-fetching.mdx:39]: Flagging (#2404)

```suggestion
Your deployed Astro site will fetch data **at build time or on the server**. In dev, you will see data fetches on component refreshes. If you need to re-fetch data multiple times client-side, use a [framework component](/en/core-concepts/framework-components/) or a [client-side script](/en/guides/client-side-scripts/) in an Astro component.
```

File: src/content/docs/en/guides/recipes.mdx [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+See guided examples of adding features to your Astro project.
+
+
+## Curated Recipes
+
+<RecipeBox icon="clipboard-check">
+  <a href="/en/guides/rss/">Add an RSS feed</a>
+</RecipeBox>
+
+<RecipeBox icon="puzzle-piece">
```
Comment (Jutanium) [inline src/content/docs/en/guides/recipes.mdx:18]: This is really three recipes. I wouldn't be afraid of having small recipes that are each their own how-to. We can adjust the UI to have parents/headings.

File: src/content/docs/en/guides/recipes.mdx [hunk 10]
```diff
@@ -0,0 +1,38 @@
+---
+title: All Recipes
```
Comment (Jutanium) [inline src/content/docs/en/guides/recipes.mdx:2]: > is it really all, if those other pages aren't here? 😅

IMO, no! I would link to their overview pages at the bottom.

**Review by Jutanium — COMMENTED** [review]
Awesome job! Starting the conversation

File: src/i18n/en/nav.ts (hunk 2 above)
Comment (sarah11918) [reply src/i18n/en/nav.ts:27]: Ooh, spicy!  :hot_pepper: 

Agreed, Routing was one I wasn't *sure* about, but I could see, until it's re-written, that MD/MDX goes in guides.

**Comment by sarah11918** [comment]
Great collaborating today, Dan! I've marked as resolved the issues that we handled today, just to keep us on track!

File: src/content/docs/en/recipes/build-forms.mdx [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+    </form>
+    ```
+    
+    :::tip
+    You can add custom validation logic, and validation logic that refers to multiple fields, using a `<script>` tag and the [Constraint Validation API](https://developer.mozilla.org/en-US/docs/Web/HTML/Constraint_validation#complex_constraints_using_the_constraint_validation_api).
+
+    To write complex validation logic more easily, you can build your form using a [frontend framework](/en/core-concepts/framework-components/) and choose a form library like [React Hook Form](https://react-hook-form.com/) or [Felte](https://felte.dev/).
+    :::
+
+4. Change the form's default data transfer `method` to `POST` to... or to avoid....(link this to what happens when submitting) The form submission will cause the browser to request the page again, with the form data stored in the [`Request` body](/en/reference/api-reference/#astrorequest).
```
Comment (sarah11918) [inline src/content/docs/en/recipes/build-forms.mdx:79]: Just calling out that I changed this up, but it still needs some deets! @Jutanium

File: src/content/docs/en/recipes/build-forms.mdx [hunk 12]
```diff
@@ -0,0 +1,200 @@
...
+    - `type="email"` also introduces validation that will only accept a valid email format.
+
+    ```astro title="src/pages/register.astro"
+    ---
+    ---
+    <h1>Register</h1>
+    <form>
+      <label>
+        Username:
+        <input type="text" name="username" required />
```
Comment (sarah11918) [inline src/content/docs/en/recipes/build-forms.mdx:59]: Calling out that I removed the length restriction on the username, and I believe I updated everything following accordingly. (Minimum length on usernames can be a sore spot for peeps with short names! The password illustrates `minlength` so I think that's fine?)

File: src/content/docs/en/recipes/build-forms.mdx [hunk 13]
```diff
@@ -0,0 +1,200 @@
...
+      </label>
+      <label>
+        Password:
+        <input type="password" name="password" required minlength="6" />
+      </label>
+      <button>Submit</button>
+    </form>
+    ```
+    
+    :::tip
```
Comment (sarah11918) [inline src/content/docs/en/recipes/build-forms.mdx:73]: Just calling attention to this tip that adding a `script` tag will make it no longer zero-JS? So, do we for sure want to add that here?

File: src/content/docs/en/recipes/build-forms.mdx [hunk 14]
```diff
@@ -0,0 +1,200 @@
+---
+title: Build HTML Forms in Astro Pages
```
Comment (sarah11918) [inline src/content/docs/en/recipes/build-forms.mdx:2]: Calling out that I changed the title and opening paragraph because "0JS" isn't the most readable, I find. So, instead I put HTML here and added something about not sending JavaScript below. 

Totally open to other suggestions here, but I really do not like how 0JS looks, so I'd really rather avoid using that.

File: src/content/docs/en/recipes/build-forms.mdx (hunk 13 above)
Comment (Jutanium) [reply src/content/docs/en/recipes/build-forms.mdx:73]: I do like having a tip that explains the case when you _do_ want JavaScript - when you need more complex validation. It _is_ an "aside".

File: src/content/docs/en/recipes/build-forms.mdx (hunk 12 above)
Comment (Jutanium) [reply src/content/docs/en/recipes/build-forms.mdx:59]: That makes sense, thanks!

File: src/content/docs/en/recipes/build-forms.mdx (hunk 14 above)
Comment (Jutanium) [reply src/content/docs/en/recipes/build-forms.mdx:2]: This works for me!

File: src/content/docs/en/recipes/build-forms.mdx (hunk 11 above)
Comment (Jutanium) [inline src/content/docs/en/recipes/build-forms.mdx:79]:
```suggestion
4. The form submission will cause the browser to request the page again. Change the form's data transfer `method` to `POST` to send the form data as part of the `Request` body, rather than as URL parameters.
```

Comment (Jutanium) [reply src/content/docs/en/recipes/build-forms.mdx:79]: https://github.com/withastro/docs/pull/2550/files/9a9030acb69fadf73725d00a7efc112b142cc18d#r1125387906

File: src/content/docs/en/core-concepts/astro-components.mdx [hunk 15]
```diff
@@ -79,22 +61,26 @@ Below the component script, sits the component template. The component template
 
 If you write plain HTML here, your component will render that HTML in any Astro page it is imported and used.
 
-However, Astro's component template syntax also supports **JavaScript expressions**, **imported components** and [**special Astro directives**](/en/reference/directives-reference/). Data and values defined (at page build time) in the component script can be used in the component template to produce dynamically-created HTML.
+However, [Astro's component template syntax](/en/core-concepts/astro-syntax/) also supports **JavaScript expressions**, **imported components** and [**special Astro directives**](/en/reference/directives-reference/). Data and values defined (during rendering) in the component script can be used in the component template to produce dynamically-created HTML.
```
[the remark this answers is not in the document]
Comment (sarah11918) [reply src/content/docs/en/core-concepts/astro-components.mdx:64]: just checking, the only thing the original provides, the one that says at build time, is the reinforcement that while these are dynamic values they're not updating ones. I'm happy to leave this out if we are comfortable losing that one extra nudge that these values are read and injected into the HTML template only once.

[the remark this answers is not in the document]
Comment (Jutanium) [reply src/content/docs/en/core-concepts/astro-components.mdx:64]: Yeah, I don't think the original provides that in an accurate/elegant way, so I think we leave it out

File: src/content/docs/en/recipes/build-forms-api.mdx [hunk 16]
```diff
@@ -0,0 +1,396 @@
...
+            </label>
+            <button>Send</button>
+          </form>
+        </template>
+        ```
+      </Fragment>
+
+    </UIFrameworkTabs>
+
+  2. Create the `post` API endpoint that will recieve the form data. Use `request.formData()` to process it. Be sure to validate the form values before you use them.
```
Comment (Yan-Thomas) [inline src/content/docs/en/recipes/build-forms-api.mdx:136]: typo!
```suggestion
  2. Create the `post` API endpoint that will receive the form data. Use `request.formData()` to process it. Be sure to validate the form values before you use them. 
```

File: src/content/docs/en/recipes/build-forms.mdx [hunk 17]
```diff
@@ -0,0 +1,200 @@
...
+      <label>
+        Password:
+        <input type="password" name="password" required minlength="6" />
+      </label>
+      <button>Submit</button>
+    </form>
+    ```
+    
+    :::tip
+    You can add custom validation logic, and validation logic that refers to multiple fields, using a `<script>` tag and the [Constraint Validation API](https://developer.mozilla.org/en-US/docs/Web/HTML/Constraint_validation#complex_constraints_using_the_constraint_validation_api).
```
Comment (Yan-Thomas) [inline src/content/docs/en/recipes/build-forms.mdx:74]: The use of "validation logic" almost back-to-back sounded a bit confusing for me, what about reworking this a bit? - Not sure that "custom validation logic that refers to multiple fields" is misleading, as in, you can't have both, so feel free to ignore and propose something else here.
```suggestion
    You can add custom validation logic that refers to multiple fields, using a `<script>` tag and the [Constraint Validation API](https://developer.mozilla.org/en-US/docs/Web/HTML/Constraint_validation#complex_constraints_using_the_constraint_validation_api).
```

File: src/content/docs/en/reference/errors/static-client-address-not-available.mdx [hunk 18]
```diff
@@ -18,7 +18,7 @@ import DontEditWarning from '~/components/DontEditWarning.astro'
 ## What went wrong?
 The `Astro.clientAddress` property is only available when [Server-side rendering](/en/guides/server-side-rendering/) is enabled.
 
-To get the user's IP address in static mode, different APIs such as [Ipify](https://www.ipify.org/) can be used in a [Client-side script](/en/core-concepts/astro-components/#client-side-scripts) or it may be possible to get the user's IP using a serverless function hosted on your hosting provider.
+To get the user's IP address in static mode, different APIs such as [Ipify](https://www.ipify.org/) can be used in a [Client-side script](/en/guides/client-side-scripts/) or it may be possible to get the user's IP using a serverless function hosted on your hosting provider.
```
Comment (Yan-Thomas) [inline src/content/docs/en/reference/errors/static-client-address-not-available.mdx:21]: Was this changed in the error.ts file in the main Astro repo as well? If not, we have to change there as well!

**Review by Yan-Thomas — COMMENTED** [review]
Great work peeps, happy to see this getting close to the finish line! 💜. 
I Added a few comments/suggestions, but nothing blocking.

File: src/content/docs/en/recipes/build-forms.mdx (hunk 17 above)
Comment (sarah11918) [reply src/content/docs/en/recipes/build-forms.mdx:74]: I like Yan's proposal! @Jutanium happy for you to approve this one, but... extra comma here because of deleting the clause, I think?

File: src/content/docs/en/reference/errors/static-client-address-not-available.mdx (hunk 18 above)
Comment (sarah11918) [reply src/content/docs/en/reference/errors/static-client-address-not-available.mdx:21]: @Yan-Thomas You're right, and the good news is, no harm in linking to the page now because that one totally exists! Making the PR there, too.

File: src/content/docs/en/recipes/build-forms.mdx (hunk 17 above)
Comment (Jutanium) [reply src/content/docs/en/recipes/build-forms.mdx:74]: I used the ol' "edit comment and commit" trick 😬

File: src/components/RecipesNav.astro [hunk 19]
```diff
@@ -0,0 +1,30 @@
...
+---
+
+<div>
+	<CardsNav
+		minimal={Astro.props.minimal}
+		links={recipes.map(({ data, slug }) => ({
+			title: data.title,
+			description: data.description,
+			href: '/' + slug.replace('en/', `${lang}/`) + '/',
+		}))}
```
Comment (delucis) [inline src/components/RecipesNav.astro:22]: This doesn’t use the translated `title` and `description`. The approach worked for other nav components because the title was always a static service name e.g. “Netlify”, “@astrojs/react”, “Storyblok” or “Gatsby”. I’ll need to tweak this logic to pull in the translated strings while still falling back to English for untranslated pages.

File: src/content/docs/en/guides/client-side-scripts.mdx [hunk 20]
```diff
@@ -8,6 +8,32 @@ i18nReady: true
 
 You can add interactivity to your Astro components without [using a UI framework](/en/core-concepts/framework-components/) like React, Svelte, Vue, etc. using standard HTML `<script>` tags. This allows you to send JavaScript to run in the browser and add functionality to your Astro components.
 
+## Client-Side Scripts
+
+Astro components support adding client-side interactivity using standard HTML `<script>` tags.
```
Comment (delucis) [inline src/content/docs/en/guides/client-side-scripts.mdx:13]: There’s a bit of repetition now that this has moved here. L9 & L13:

> You can add interactivity to your Astro components […] using standard HTML `<script>` tags.

> Astro components support adding client-side interactivity using standard HTML `<script>` tags.

I’ll also call out that this content used to be in astro-components as a kind of hint that this was possible, linking here. This means that a) now that it’s on this page, it’s repeats some stuff later on this page and b) we no longer call-out client-side scripts (or styles!) are possible anywhere in either the `astro-components` or the `astro-syntax` page, which doesn’t seem ideal.

File: src/content/docs/en/reference/cli-reference.mdx [hunk 21]
```diff
@@ -1,12 +1,143 @@
...
 i18nReady: true
 ---
 import Since from '~/components/Since.astro';
 import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro'
 
 
 
+You can use the Command-Line Interface (CLI) provided by Astro to develop, build, and preview your project from a terminal window.
```
Comment (delucis) [inline src/content/docs/en/reference/cli-reference.mdx:10]: The additions to this page are really helpful! Happy to see these get a clear explanation.

File: src/i18n/en/nav.ts [hunk 22]
```diff
@@ -13,71 +13,81 @@ export default [
...
 		text: 'Server-side Rendering (SSR)',
 		slug: 'guides/server-side-rendering',
 		key: 'guides/server-side-rendering',
 	},
-	{ text: 'Authoring Content', slug: 'guides/content', key: 'guides/content' },
 	{
-		text: 'Content Collections',
-		slug: 'guides/content-collections',
-		key: 'guides/content-collections',
+		text: 'Scripts & Event Handling',
```
Comment (delucis) [inline src/i18n/en/nav.ts:72]: Personal preference would be to keep this close to the CSS & Styling sidebar entry as CSS + JS are two related types of content.

File: src/i18n/en/nav.ts [hunk 23]
```diff
@@ -13,71 +13,81 @@ export default [
...
+		slug: 'guides/client-side-scripts',
+		key: 'guides/client-side-scripts',
 	},
-	{ text: 'Connecting a CMS', slug: 'guides/cms', key: 'guides/cms' },
+	{ text: 'Markdown & MDX', slug: 'guides/markdown-content', key: 'guides/markdown-content' },
```
Comment (delucis) [inline src/i18n/en/nav.ts:76]: Similarly, feels odd that MD/MDX is not next to Content Collections.

File: src/i18n/en/nav.ts [hunk 24]
```diff
@@ -13,71 +13,81 @@ export default [
...
-		key: 'core-concepts/sharing-state',
-	},
-	{ text: 'RSS', slug: 'guides/rss', key: 'guides/rss' },
+	{ text: 'Imports', slug: 'guides/imports', key: 'guides/imports' },
+	{ text: 'Endpoints', slug: 'core-concepts/endpoints', key: 'core-concepts/endpoints' },
+	{ text: 'Data Fetching', slug: 'guides/data-fetching', key: 'guides/data-fetching' },
 	{ text: 'Testing', slug: 'guides/testing', key: 'guides/testing' },
+	{ text: 'Troubleshooting', slug: 'guides/troubleshooting', key: 'guides/troubleshooting' },
+
+	// { text: 'Authoring Content', slug: 'guides/content', key: 'guides/content' },
```
Comment (delucis) [inline src/i18n/en/nav.ts:87]: If we’re happy with the nav bar, would be good to clean up old comments here.

**Review by delucis — COMMENTED** [review]
These changes are super positive! Already feels like such an organisational improvement as well as giving us scope to grow in exciting ways.

Mentioned on Discord that I’d suggest pulling @Jutanium’s big _new_ recipes about forms into a separate dedicated PR to give us time and space to review those independently.

File: src/content/docs/en/guides/client-side-scripts.mdx (hunk 20 above)
Comment (sarah11918) [reply src/content/docs/en/guides/client-side-scripts.mdx]: I believe I addressed this!

**Review by delucis — APPROVED** [review]
I think this PR pot’s been bubbling long enough and these recipes are ready to serve! 🚀
